src/js/components/modals/JobFormModal.js

```diff
@@ -145,27 +152,87 @@ class JobFormModal extends mixin(StoreMixin) {
   }
 
   handleInputModeToggle() {
-    this.setState({jsonMode: !this.state.jsonMode});
+    this.setState({jsonMode: !this.state.jsonMode, errorMessage: null});
   }
 
   handleJSONEditorChange(jsonDefinition) {
-    try {
-      let job = new Job(JSON.parse(jsonDefinition));
-      this.setState({errorMessage: null, job});
-    } catch (error) {
-      // TODO: DCOS-7734 Handle error
-    }
+
+    // Keep the new JSON string in the JSON definition
+    // Validation will occur upon submission of the form
+    this.setState({
+      errorMessage: null,
+      jsonDefinition
+    });
+
   }
 
   handleSubmit() {
+    let {jsonMode} = this.state;
     let {isEdit, job} = this.props;
-    let jobSpec = JobUtil.createJobSpecFromJob(this.state.job);
 
-    if (!isEdit) {
-      MetronomeStore.createJob(jobSpec);
+    if (jsonMode) {
+      try {
+
+        // Parse JSON and create job object
+        let job = new Job(JSON.parse(this.state.jsonDefinition));
```

Comment:
You are correct. However this whole branch will be re-written soon, since I am working on a client-side JSON validator.